patches/minecraft/net/minecraft/client/player/LocalPlayer.java.patch

```diff
@@ -37,6 +65,15 @@
              this.f_108617_.m_104955_(new ServerboundPlayerCommandPacket(this, ServerboundPlayerCommandPacket.Action.START_FALL_FLYING));
           }
        }
+ 
+       this.f_108614_ = this.m_21255_();
+-      if (this.m_20069_() && this.f_108618_.f_108573_ && this.m_6129_()) {
+-         this.m_21208_();
++      if ((this.m_20069_() || this.isInCustomFluid()) && this.f_108618_.f_108573_ && this.m_6129_()) {
++         this.getTouchingCustomFluid().sink(this);
```

Comment:
Given water is not a custom fluid I believe this breaks vanilla functionality.

Reply:
Unless it is a custom fluid with overriden behaviour, it'll use the same "sinking" logic as vanilla.

Original: this.goDownInWater
```java
protected void goDownInWater() {
      this.setDeltaMovement(this.getDeltaMovement().add(0.0D, (double)-0.04F * this.getAttribute(net.minecraftforge.common.ForgeMod.SWIM_SPEED.get()).getValue(), 0.0D));
   }
   ```
   
   Patch: this.getTouchingCustomFluid().sink(this)
   ```java
       default void sink(LivingEntity entity)
    {
        self().getType().sink(self(), entity);
    }
    
    Which calls
    
        default void sink(FluidState state, LivingEntity entity)
    {
        entity.setDeltaMovement(entity.getDeltaMovement().add(0.0D, (double)-0.04F * entity.getAttribute(ForgeMod.SWIM_SPEED.get()).getValue(), 0.0D));
    }
    ```